factory/utils.py

```diff
@@ -16,6 +16,14 @@ def import_object(module_name, attribute_name):
     return getattr(module, attribute_name)
 
 
+def resolve_type(type_or_path):
+    if isinstance(type_or_path, type):
+        return type_or_path
+    if not (isinstance(type_or_path, str) and '.' in type_or_path):
+        raise ValueError("Must receive either an object  or the fully qualified path")
```

Comment:
There is an extra space between "object" and "or".

```suggestion
        raise ValueError("Must receive either an object or the fully qualified path")
```